pkg/scheduler/api/v1/types.go

```diff
@@ -158,6 +161,28 @@ type ExtenderArgs struct {
 	NodeNames *[]string `json:"nodenames,omitempty"`
 }
 
+// ExtenderPreemptionResult represents the result returned by preemption phase of extender.
+type ExtenderPreemptionResult map[string]*Victims
+
+// ExtenderPreemptionArgs represents the arguments needed by the extender to preempt pods on nodes.
+type ExtenderPreemptionArgs struct {
+	// Pod being scheduled
+	Pod *apiv1.Pod `json:"pod"`
+	// Victims map generated by scheduler preemption phase
+	NodeToVictims map[string]*Victims `json:"nodeToPods"`
+	// PodDisruptionBudget list
+	// +optional
+	Pdbs []*policy.PodDisruptionBudget `json:"pdbs,omitempty"`
+}
+
+// Victims represents:
+//   pods:  a group of pods expected to be preempted.
+//   numPDBViolations: the count of violations of PodDisruptionBudget
+type Victims struct {
+	Pods             []*apiv1.Pod `json:"pods"`
```

Comment:
Should this be names of pods? Again, assumption that extender gets the pods from apiserver.

Reply:
Name of pod is not unique. We need namespace at least.

Reply:
Yes, size of the message becomes an issue when we included the whole struct. Pod Name, Namespace, UUID should suffice. UUID is important to keep the extender in sync.

Reply:
Not very sure if this is overkill, cc @bsalamat any experience of victim list size to see if we need to pass pod name list instead?

Reply:
Sending only Pod identifiers (name, namspace, UUID) is fine with me. Generally speaking, we don't expect the list of victims to be long, probably only several pods, but given that many nodes may be chosen as possible preemption candidates in larger clusters, limiting the information we pass can help improve performance.